go-controller/pkg/cluster/ha_master.go

```diff
@@ -0,0 +1,270 @@
+		if hacluster.nodeName == identity {
+			_ = hacluster.ConfigureAsActive(identity)
+		} else {
+			_ = hacluster.ConfigureAsStandby(identity)
+		}
+	}
+
+	lec := leaderelection.LeaderElectionConfig{
+		Lock:          rl,
+		LeaseDuration: leaseDuration,
+		RenewDeadline: renewDeadline,
+		RetryPeriod:   retryPeriod,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {},
+			OnStoppedLeading: func() {},
+			OnNewLeader:      HAClusterNewLeader,
+		},
+	}
+
+	go leaderelection.RunOrDie(context.Background(), lec)
+
+	return nil
+}
+
+// ConfigureAsActive configures the node as active.
+func (hacluster *OvnHAMasterController) ConfigureAsActive(masterNodeName string) error {
+	if hacluster.manageDBServers {
+		// Find the endpoint for the service
+		ep, err := hacluster.cluster.Kube.GetEndpoint(config.Kubernetes.OVNConfigNamespace, ovnkubeDbEp)
+		createEp := true
+		if err == nil {
+			if ep.Name == ovnkubeDbEp {
+				createEp = false
+			}
+		}
+
+		const (
+			nbPort int32 = 6641
+			sbPort int32 = 6642
+		)
+
+		ovndbEp := v1.Endpoints{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: config.Kubernetes.OVNConfigNamespace,
+				Name:      ovnkubeDbEp,
+			},
+			Subsets: []v1.EndpointSubset{
+				{
+					Addresses: []v1.EndpointAddress{
+						{IP: config.Kubernetes.OVNKubePodIP},
+					},
+					Ports: []v1.EndpointPort{
+						{
+							Name: "north",
+							Port: nbPort,
+						},
+						{
+							Name: "south",
+							Port: sbPort,
+						},
+					},
+				},
+			},
+		}
+
+		if createEp {
+			_, err := hacluster.cluster.Kube.CreateEndpoint(config.Kubernetes.OVNConfigNamespace, &ovndbEp)
+			if err != nil {
+				//TODO: Handle this situation properly.
+				logrus.Infof("%s Endpount Create failed", ovnkubeDbEp)
+			}
+		} else {
+			_, err := hacluster.cluster.Kube.UpdateEndpoint(config.Kubernetes.OVNConfigNamespace, &ovndbEp)
+			if err != nil {
+				//TODO: Handle this situation properly.
+				logrus.Infof("%s Endpount Update failed", ovnkubeDbEp)
+			}
+		}
+
+		// Promote the OVN DB servers.
+		_ = hacluster.PromoteOVNDbs(strconv.Itoa(int(nbPort)), strconv.Itoa(int(sbPort)))
+	}
+
+	// run the cluster controller to init the master
+	err := hacluster.cluster.StartClusterMaster(hacluster.nodeName)
+	if err != nil {
+		logrus.Errorf(err.Error())
+		panic(err.Error())
+	}
+
+	if err := hacluster.ovnController.Run(); err != nil {
+		logrus.Errorf(err.Error())
+		panic(err.Error())
+	}
+
+	return nil
+}
+
+// ConfigureAsStandby configures the node as standby
+func (hacluster *OvnHAMasterController) ConfigureAsStandby(masterNodeName string) error {
+	if hacluster.manageDBServers {
+		// Get the ovndb endpoint.
+		ep, err := hacluster.cluster.Kube.GetEndpoint(config.Kubernetes.OVNConfigNamespace, ovnkubeDbEp)
+		if err != nil {
+			logrus.Infof("[%s] Endpount Not found. Need to handle this", ovnkubeDbEp)
+			return err
+		}
+
+		// Get the master ip
+		masterIP := ep.Subsets[0].Addresses[0].IP
```

Comment:
@numansiddique can you extract a common endpoint parser from this and updateOVNConfig() and put that into pkg/cluster/cluster.go and call it from both here and node.go?

Reply:
Sure.

Reply:
Ack. Done